### PR TITLE
Initial changes for TeensyEyes

### PR DESCRIPTION
**Pull request by MichaelMeissner**
These are 3 commits:

1) Add more things to .gitignore (emacs and patch under Linux leave other temporary files around)

2) Add final newline to the files that need it (some of the Linux source tools don't work as well if a file does not have a final newline).

3) Add files to build under Arduino.

One other change that you need to do to build under Arduino is create a dummy .ino file in the src directory.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,4 +33,14 @@ cmake-*
 *.app
 
 # PlatformIO
-.pio
```
Comment (chrismiller) [inline .gitignore]: The *.diff exclude should be in a new section, not appended to the end of .pio

**Review by chrismiller — APPROVED** [review]
Thanks. As a habit I tend not to include trailing new lines (for no particular reason), so hopefully more of these won't slip in again accidentally. I don't think I have a good way to enforce this other than writing a pre-commit hook or something. [Though more generally I'd argue that if tools can't handle valid code, it's the tools that need fixing, not the code!]

**Review by chrismiller — DISMISSED** [review]

**Review by chrismiller — COMMENTED** [review]
Hmm I'm not quite sure what to do about this one. The problem with this is that when eyes are regenerated with different parameters, a different set of *.h/*.cpp file names will be generated too, which will break a lot of these #includes :(

One possible solution might be for the Python codegen to also generate these build-*.cpp files, so they don't need editing manually. Are you certain these inclusions are really needed though? A quick search seems to indicate that Arduino IDE does have some recursive build capabilities, e.g. https://github.com/arduino/Arduino/issues/4821. Is the source of the problem actually something to do with the includes (#include "polarAngle_240.h" etc) in each eye source file (e.g. cat.h, brown.h, ...) not resolving correctly for you? Because it seems like all the other files (including cat.h, brown.h, ... themselves) _are_ being found and compiled OK for you, even though they are also in subdirectories.

**Comment by MichaelMeissner** [comment]
Note, I did merge in my pin defaults in config.h which now leaves config.h in conflict.  Obviously keep your definitions, and I'll keep mine in my tree.

But I don't know enough of the git stuff to do that.  When I'm using git within GCC, we don't use pull requests, etc.  Instead, we mail in the patch, and when it is approved, the person doing the patch does the commit.

In terms of the *.cpp files in src/eyes/240x240, I think the issue is if this was structured as a library, the Arduino builder will build the .cpp files in sub-directories.  But since it is a directory with an .ino file, it does not descend into sub-dirs.  Hence, it will compile the various eye files, but the linker will give errors that polarAngle_240, etc. don't exist.

**Pull request closed without merging**